v0/destinations/snapchat_conversion/transform.js

```diff
@@ -185,27 +185,50 @@ function trackResponseBuilder(message, { Config }) {
     payload.event_tag = message.properties.event_tag;
   }
 
-  payload.hashed_email = getHashedValue(
-    message?.context?.traits?.email
-      ?.trim()
-      ?.toString()
-      ?.toLowerCase()
-  );
-  payload.hashed_phone_number = getHashedValue(
-    getNormalizedPhoneNumber(message)
-      ?.toString()
-      ?.toLowerCase()
-  );
-  payload.user_agent = message?.context?.userAgent?.toString()?.toLowerCase();
-  payload.hashed_ip_address = getHashedValue(
-    message?.context?.ip?.toString()?.toLowerCase()
-  );
-  (hashed_mobile_ad_id = getHashedValue(
-    message?.context?.idfa?.toString()?.toLowerCase()
-  )),
-    (hashed_idfv = getHashedValue(
-      message?.context?.idfv?.toString()?.toLowerCase()
-    ));
+  const email = getFieldValueFromMessage(message, "email");
+  if (email) {
+    payload.hashed_email = getHashedValue(
+      email
+        .toString()
+        .toLowerCase()
+        .trim()
+    );
+  }
+  const phone = getNormalizedPhoneNumber(message);
+  if (phone) {
+    payload.hashed_phone_number = getHashedValue(
+      phone
+        .toString()
+        .toLowerCase()
+        .trim()
+    );
+  }
+  const ip = message.context?.ip || message.request_ip;
+  if (ip) {
+    payload.hashed_ip_address = getHashedValue(
+      ip
+        .toString()
+        .toLowerCase()
+        .trim()
+    );
+  }
+  // only in case of ios platform this is required
+  if (
+    message.context?.device?.type?.toLowerCase() === "ios" &&
```

Comment:
We should add the validation for all apple devices instead of just "ios". We have an `isAppleFamily(..)` util function for it.

Reply:
Make sense